src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -162,6 +162,19 @@ public async Task<FormulaValue> EvalAsync(string expressionText, CancellationTok
             return await check.Expression.EvalAsync(parameters, cancel);
         }
 
+        public (IEnumerable<ExpressionError>, IEnumerable<FunctionInfo>) DefineFunctions(string script)
+        {
+            var parsedUDFS = new Core.Syntax.ParsedUDFs(script);
+            var result = parsedUDFS.GetParsed();
+
+            var udfDefinitions = result.UDFs.Select(udf => new UDFDefinition(

```

Comment:
Are we checking to see if the actual return type matches the stated return type?

Reply:
done